ambari-server/src/main/resources/stack-hooks/before-START/scripts/params.py

```diff
@@ -269,7 +269,7 @@
   
 ambari_libs_dir = "/var/lib/ambari-agent/lib"
 is_webhdfs_enabled = config['configurations']['hdfs-site']['dfs.webhdfs.enabled']
-default_fs = config['configurations']['core-site']['fs.defaultFS']
+default_fs = default('/configurations/core-site/fs.defaultFS',None)
```

Comment:
Can you please explain the need for this change? Please include steps to reproduce the problem it is trying to solve.

Reply:
Step as follows:
when deploy a cluster with only 'zookeeper'，zookeeper started failed。

Traceback (most recent call last):
File "/var/lib/ambari-agent/cache/stacks/HDP/2.0.6/hooks/before-START/scripts/hook.py", line 40, in <module>
BeforeStartHook().execute()
File "/usr/lib/python2.6/site-packages/resource_management/libraries/script/script.py", line 314, in execute
method(env)
File "/var/lib/ambari-agent/cache/stacks/HDP/2.0.6/hooks/before-START/scripts/hook.py", line 28, in hook
import params
File "/var/lib/ambari-agent/cache/stacks/HDP/2.0.6/hooks/before-START/scripts/params.py", line 320, in <module>
if namenode_rpc:
File "/usr/lib/python2.6/site-packages/resource_management/libraries/script/config_dictionary.py", line 73, in _getattr_
raise Fail("Configuration parameter '" + self.name + "' was not found in configurations dictionary!")
resource_management.core.exceptions.Fail: Configuration parameter 'core-site' was not found in configurations dictionary!
Error: Error: Unable to run the custom hook script ['/usr/bin/python', '/var/lib/ambari-agent/cache/stacks/HDP/2.0.6/hooks/before-START/scripts/hook.py', 'START', '/var/lib/ambari-agent/data/command-31.json', '/var/lib/ambari-agent/cache/stacks/HDP/2.0.6/hooks/before-START', '/var/lib/ambari-agent/data/structured-out-31.json', 'INFO', '/var/lib/ambari-agent/tmp', 'PROTOCOL_TLSv1']

Reply:
Thanks, the traceback helps a lot.  This problem is already fixed by 108ad9b4a0d and c35a834d7bf:

https://github.com/apache/ambari/blob/79504f74e83164a9c70713ea54cfc2f257d5be04/ambari-server/src/main/resources/stack-hooks/before-START/scripts/params.py#L336-L337

and

https://github.com/apache/ambari/blob/2019a5c0d824c73167535ac7091fc1af765c4826/ambari-server/src/main/resources/stacks/HDP/2.0.6/hooks/before-START/scripts/params.py#L330-L331